commit 3f791c60cccd506e05ac36d895969618a93014ed
Author: Justin Stitt <justinstitt@google.com>
Date:   Wed Oct 18 21:50:01 2023 +0000

    wifi: wlcore: main: replace deprecated strncpy with strscpy
    
    strncpy() is deprecated for use on NUL-terminated destination strings
    [1] and as such we should prefer more robust and less ambiguous string
    interfaces.
    
    We expect fw_version strings to be NUL-terminated based on other similar
    assignments:
    
    wireless/broadcom/brcm80211/brcmsmac/main.c
    7867:           snprintf(wlc->wiphy->fw_version,
    7868:                    sizeof(wlc->wiphy->fw_version), "%u.%u", rev, patch);
    
    wireless/broadcom/b43legacy/main.c
    1765:   snprintf(wiphy->fw_version, sizeof(wiphy->fw_version), "%u.%u",
    
    wireless/broadcom/b43/main.c
    2730:   snprintf(wiphy->fw_version, sizeof(wiphy->fw_version), "%u.%u",
    
    wireless/intel/iwlwifi/dvm/main.c
    1465:   snprintf(priv->hw->wiphy->fw_version,
    1466:            sizeof(priv->hw->wiphy->fw_version),
    
    wireless/intel/ipw2x00/ipw2100.c
    5905:   snprintf(info->fw_version, sizeof(info->fw_version), "%s:%d:%s",
    
    Based on this, NUL-padding is not required.
    
    A suitable replacement is `strscpy` [2] due to the fact that it
    guarantees NUL-termination on the destination buffer without
    unnecessarily NUL-padding.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Signed-off-by: Kalle Valo <kvalo@kernel.org>
    Link: https://lore.kernel.org/r/20231018-strncpy-drivers-net-wireless-ti-wlcore-main-c-v1-1-1b1055f482a1@google.com

diff --git a/drivers/net/wireless/ti/wlcore/main.c b/drivers/net/wireless/ti/wlcore/main.c
index 448c478a0fdd..fb9ed97774c7 100644
--- a/drivers/net/wireless/ti/wlcore/main.c
+++ b/drivers/net/wireless/ti/wlcore/main.c
@@ -1126,7 +1126,7 @@ int wl1271_plt_start(struct wl1271 *wl, const enum plt_mode plt_mode)
 
 		/* update hw/fw version info in wiphy struct */
 		wiphy->hw_version = wl->chip.id;
-		strncpy(wiphy->fw_version, wl->chip.fw_ver_str,
+		strscpy(wiphy->fw_version, wl->chip.fw_ver_str,
 			sizeof(wiphy->fw_version));
 
 		goto out;
@@ -2344,7 +2344,7 @@ static int wl12xx_init_fw(struct wl1271 *wl)
 
 	/* update hw/fw version info in wiphy struct */
 	wiphy->hw_version = wl->chip.id;
-	strncpy(wiphy->fw_version, wl->chip.fw_ver_str,
+	strscpy(wiphy->fw_version, wl->chip.fw_ver_str,
 		sizeof(wiphy->fw_version));
 
 	/*